src/libs/actions/Report.js

```diff
@@ -1081,9 +1081,10 @@ function fetchAllReports(
  * @param {File} [file]
  */
 function addAction(reportID, text, file) {
-    // Convert the comment from MD into HTML because that's how it is stored in the database
+    // For comments shorter than 10k chars, convert the comment from MD into HTML because that's how it is stored in the database
+    // For longer comments, skip parsing and display plaintext for performance reasons. It takes over 40s to parse a 100k long string!!
     const parser = new ExpensiMark();
-    const commentText = parser.replace(text);
+    const commentText = text.length < 10000 ? parser.replace(text) : text;
```

Comment:
**question:** Does this mean we will not render markdown for these?

Reply:
10000 characters feels kind of magic. How did we land on this? Also what about a solution that accounts for the restrictions in Auth and this issue? e.g. could we chunk this into a few different messages so that it stays under the limit?

Reply:
>question: Does this mean we will not render markdown for these?

Correct!

>10000 characters feels kind of magic. How did we land on this? Also what about a solution that accounts for the restrictions in Auth and this issue? e.g. could we chunk this into a few different messages so that it stays under the limit?

10k came from the timing in my tests [here](https://github.com/Expensify/Expensify/issues/205451#issuecomment-1102845551), e.g. 20k chars took 2.2s to parse. 

I tried chunking, i.e. calling `addAction` multiple times with chunks of 1k, 5k, 10k and 20k chars, but that didn't seem to have any effect on the performance. I _think_ that the regex parsing is at least a O(n) operation, so breaking it down into chunks wouldn't matter when parsing `n` or `m*n/m` chars. Having said that, maybe I'm missing something in your suggestion?

Reply:
Did we perform those tests on a development or production build? 
Should we expect this work to be relatively consistent from one machine to the next?

Reply:
>Did we perform those tests on a development or production build?

That was on a development build.

>Should we expect this work to be relatively consistent from one machine to the next?

The amount of work (number of operations) should be the same across machines, but the timing will vary. A prod build running on a mobile device would probably be the best benchmark to use, although I think it's more common for users to send large messages over web/desktop.

Reply:
So if we put in a large markdown comment it will work as text, but not get converted. That seems kind of broken, but also unsure if it's worth optimizing right now.

>Having said that, maybe I'm missing something in your suggestion?

I see. So chunking didn't improve the performance? Where is the performance bottleneck exactly - in the JS thread? If we are having an issue with the performance of our markdown parser the first thoughts that come to mind are...

- Can we make the parser more efficient?
- If not, can we move the work the parser has to do onto a different thread (assuming our issue is that it's blocking the main JS)?

Separately, I took a look at the original issue which mentions stuff about how someone could "crash" another client by sending a message and unclear how it is related to parsing when we are _adding_ a comment or how this change will fix that.

Anyways, just curious about this, thanks!

Reply:
>So if we put in a large markdown comment it will work as text, but not get converted. That seems kind of broken, but also unsure if it's worth optimizing right now.

Agreed. With the current solution, we could limit the number of characters to 10k and not disable markdown.

>So chunking didn't improve the performance? 

I didn't see any improvements in performance.

>Where is the performance bottleneck exactly - in the JS thread?

Correct!

>Can we make the parser more efficient?

Probably and I considered it, but the more I looked into it the more of a rabbit hole it became 😅  specially considering this is a bit of an edge case. We could look at some external parsing library (assuming that it would be faster than ours).

>If not, can we move the work the parser has to do onto a different thread (assuming our issue is that it's blocking the main JS)?

This looks like a possibility. It would stop blocking the JS thread, but wouldn't it still take a long time to parse the message?

>Separately, I took a look at the original issue which mentions stuff about how someone could "crash" another client by sending a message and unclear how it is related to parsing when we are adding a comment or how this change will fix that.

Ah yea, but that assumed that sending such a large message to a user would succeed. Auth actually returns a `402 Maximum size exceeded` error to that API call.

Reply:
>This looks like a possibility. It would stop blocking the JS thread, but wouldn't it still take a long time to parse the message?

Yeah I'm not sure if this is a _great_ idea or even possible, but at a high level it seems the user could at least do other things while the parsing happens.

>Ah yea, but that assumed that sending such a large message to a user would succeed.

<img width="482" alt="2022-04-20_14-54-08" src="https://user-images.githubusercontent.com/32969087/164350330-29345718-0730-4591-b7c9-5eab3a4a7c9e.png">

Ok got it. So that's not our problem then. And we have an issue with the markdown parser performance. Thanks!

Not necessarily against the idea to prevent large comments if there is no way to support them, but the parser will still be slow if we do this - so wondering if we need to look to improve the performance of the parsing in general or what exactly is happening there. I think it's not the first time I heard it was slow, but can't remember the exact context.

Reply:
Yea, I'm gonna look into improving the performance next 👍

Reply:
Ok, I created an issue to investigate our parser performance and the slow regex test - https://github.com/Expensify/Expensify/issues/208027.